tools/transform.cpp

```diff
@@ -746,12 +511,63 @@ static void calculateAndInitConstants(Transform &t) {
   bool has_ptr_load = false;
   does_sub_byte_access = false;
 
+
+  for (auto fn : { &t.src, &t.tgt }) {
+    unsigned &num_locals = fn == &t.src ? num_locals_src : num_locals_tgt;
+    uint64_t &max_gep    = fn == &t.src ? max_gep_src : max_gep_tgt;
+
+    for (auto BB : fn->getBBs()) {
+      for (auto &i : BB->instrs()) {
+        if (returns_local(i))
+          ++num_locals;
+        else
+          num_max_nonlocals_inst += returns_nonlocal(i);
+
+        if (auto *mi = dynamic_cast<const MemInstr *>(&i)) {
+          max_alloc_size  = max(max_alloc_size, mi->getMaxAllocSize());
+          max_access_size = max(max_access_size, mi->getMaxAccessSize());
+          max_gep         = add_saturate(max_gep, mi->getMaxGEPOffset());
```

Comment:
I think this needs to add alloc_size & access size as well.
Otherwise the code doesn't seem equivalent to what we had before. I'm surprise no test fails (it should!). When doing loads, allocs, etc, we need offsets to be big enough to accommodate the UB constraints.

Reply:
It's done at line 634 and below.
Maybe the name of max_gep was a bit confusing (there is max_geps as well), so I changed it

Reply:
the alloc size is not considered AFAICT

Reply:
I think max_alloc_size was reflected as well:
```
  // we need an extra bit because 1st bit of size is always 0
  bits_size_t = ilog2_ceil(max_alloc_size, true);
  bits_size_t = min(max(bits_for_offset, bits_size_t)+1, bits_program_pointer);
```
`max_alloc_size` was reflected to `bits_size_t` but not to `bits_for_offset` in the original code.